src/recap

```diff
@@ -146,492 +134,440 @@ print_usage() {
-  		cat $PSTREE_FILE >> $TEMPDIR/report.log
-		echo "-----END PSTREE REPORT-----" >> $TEMPDIR/report.log
-		echo " " >> $TEMPDIR/report.log
-	fi
-
-	if [ -f $RESOURCES_FILE ] && [ $USERESOURCES = "yes" -o $USERESOURCES = "YES" ]
-	then
-		echo "-----BEGIN RESOURCES REPORT-----" >> $TEMPDIR/report.log
-		cat $RESOURCES_FILE >> $TEMPDIR/report.log
-		echo "-----END RESOURCES REPORT-----" >> $TEMPDIR/report.log
-		echo " " >> $TEMPDIR/report.log
-	fi
-
-	if [ -f $NETSTAT_FILE ] && [ $USENETSTAT = "yes" -o $USENETSTAT = "YES" ]
-	then
-		echo "-----BEGIN NETSTAT REPORT-----" >> $TEMPDIR/report.log
-		cat $NETSTAT_FILE >> $TEMPDIR/report.log
-		echo "-----END RESOURCES REPORT-----" >> $TEMPDIR/report.log
-		echo " " >> $TEMPDIR/report.log
-	fi
-
-	if [ -f $MYSQL_FILE ] && [ $USEMYSQL = "yes" -o $USEMYSQL = "YES" ]
-	then
-		echo "-----BEGIN MYSQL REPORT-----" >> $TEMPDIR/report.log
-		cat $MYSQL_FILE >> $TEMPDIR/report.log
-		echo "-----END RESOURCES REPORT-----" >> $TEMPDIR/report.log
-		echo " " >> $TEMPDIR/report.log
-	fi
-
-  	# send email summary to MAILTO address
-	if [ -f $TEMPDIR/report.log ]
-	then
-		cat $TEMPDIR/report.log | mail -s "System Monitor Report - $HOSTNAME" $MAILTO
-
-  		# clean up after ourselves 
-		rm -f $TEMPDIR/report.log
-		rm -rf $TEMPDIR
-	fi
-}
-
-# manage ps report
-run_ps_report() {
-	recaplock
-
-	if [ $SNAPSHOT = "yes" ]
-	then
-		PS_FILE="$SNAPSHOTSDIR/ps.log_snapshot_$DATE";
-	fi
+  # create a temp directory
+  local TEMPDIR=$( mktemp -d "/tmp/rs.XXXXXXXXXX" )
+  local REPORT="${TEMPDIR}/report.log"
+  local ITEM_FILE=""
+  local ITEM_USE=""
+
+  # echo report files into temporary file for mailing
+  for item in ${BACKUP_ITEMS}; do
+    ITEM_FILE="${BASEDIR}/${item}_${LOG_SUFFIX}.log"
+    ITEM_USE=$( eval "echo \${USE${item^^}}" )
+    if [[ -f "${ITEM_FILE}" && ${ITEM_USE,,} == "yes" ]]; then
+      echo "-----BEGIN ${item^^} REPORT-----" >> "${REPORT}"
+      cat "${ITEM_FILE}" >> "${REPORT}"
+      echo "-----END ${item^^} REPORT-----" >> "${REPORT}"
+      echo " " >> "${REPORT}"
+    fi
+  done
 
-	create_output_file $PS_FILE
-	check_output_file $PS_FILE
-	print_ps
+  # send email summary to MAILTO address
+  if [[ -r "${REPORT}" ]]; then
+    cat "${REPORT}" | mail -s "System Monitor Report - ${HOSTNAME}" "${MAILTO}"
+    # Remove the temporary files
+    rm -rf "${TEMPDIR}"
+  fi
 }
 
-# manage pstree report
-run_pstree_report() {
-	if [ $SNAPSHOT = "yes" ]
-	then
-		PSTREE_FILE="$SNAPSHOTSDIR/pstree.log_snapshot_$DATE";
-	fi
-
-	create_output_file $PSTREE_FILE
-	check_output_file $PSTREE_FILE
-	print_pstree
-}
+# manage item report
+run_item_report() {
+  local item="$1"
+  local ITEM_FILE="${BASEDIR}/${item}_${LOG_SUFFIX}.log"
 
-# manage fdisk report
-run_fdisk_report() {
-	if [ $SNAPSHOT = "yes" ]
-	then
-		FDISK_FILE="$SNAPSHOTSDIR/fdisk.log_snapshot_$DATE";
-	fi
+  if [[ ${SNAPSHOT,,} == "yes" ]]; then
+    ITEM_FILE="${SNAPSHOTSDIR}/${item}.log_snapshot_${DATE}"
```

Comment:
SNAPSHOTSDIR only existed inside the create_output_dirs function.  It's no longer a global variable, after your changes.